### PR TITLE
If a SPDM message size is greater than the transmit buffer siz…

### PR DESCRIPTION
**Pull request by Zhiqiang520**
…e of the sending SPDM endpoint, then it is also a large SPDM message.

Fix: #1865

### CODE REVIEW

File: library/spdm_responder_lib/libspdm_rsp_receive_send.c [hunk 1]
```diff
@@ -593,8 +593,12 @@ libspdm_return_t libspdm_build_response(void *spdm_context, const uint32_t *sess
         }
     }
 
-    if ((context->connection_info.capability.data_transfer_size != 0) &&
-        (my_response_size > context->connection_info.capability.data_transfer_size) &&
+    /* large SPDM message is the SPDM message whose size is greater than the DataTransferSize of the receiving
+     * SPDM endpoint or greater than the transmit buffer size of the sending SPDM endpoint */
+    if ((((context->connection_info.capability.data_transfer_size != 0) &&
+          (my_response_size > context->connection_info.capability.data_transfer_size)) ||
+         ((context->local_context.capability.data_transfer_size != 0) &&
```
Comment (jyao1) [inline library/spdm_responder_lib/libspdm_rsp_receive_send.c:600]: I dont understand the logic.
Why you check local receive buffer size?

Comment (Zhiqiang520) [reply library/spdm_responder_lib/libspdm_rsp_receive_send.c:600]: currently one macro `LIBSPDM_SENDER_RECEIVE_BUFFER_SIZE` defined both of the receiver buffer size and the sender buffer size for one same SPDM endpoint, so I take the local receive buffer size as its send buffer size here.

Comment (jyao1) [reply library/spdm_responder_lib/libspdm_rsp_receive_send.c]: now we have LIBSPDM_SENDER_DATA_TRANSFER_SIZE and LIBSPDM_RECEIVER_DATA_TRANSFER_SIZE

Comment (Zhiqiang520) [reply library/spdm_responder_lib/libspdm_rsp_receive_send.c]: I have updated the pull request, please check.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**